index.bs

```diff
@@ -119,105 +133,462 @@ A device is [=capable of supporting=] the depth sensing feature if the device ex
+
+  float getDepthInMeters(unsigned long column, unsigned long row);
+};
 </script>
 
-The {{XRDepthInformation/data}} attribute contains depth buffer information in raw format, suitable for uploading to a WebGL texture. The data is stored in column-major format, with each 2-byte entry containing unsigned integer corresponding to distance from the [=XRDepthInformation/view=]'s near plane, in millimeters.
+The {{XRCPUDepthInformation/data}} attribute contains depth buffer information in raw format, suitable for uploading to a WebGL texture if needed. The data is stored in column-major format, without padding, with each entry containing unsigned integer corresponding to distance from the [=XRDepthInformation/view=]'s near plane, in unspecified units. The size of each data entry and the type is determined by {{XRSession/depthDataFormat}}. The values can be converted from unspecified units to meters by multiplying them by {{XRDepthInformation/rawValueToMeters}}. When accessed, the algorithm to <a lt="attempting to access members">access a member</a> MUST be run.
 
-Issue(immersive-web/depth-sensing#6): Accessing raw data could potentially need to have a way of knowing what is the underlying data type and how to convert from the numeric value to a value expressed in meters. This is to ensure that we do not force the implementers to convert the data they are able to obtain from the underlying XR devices.
+Note: Applications SHOULD NOT attempt to change the contents of {{XRCPUDepthInformation/data}} array as this can lead to incorrect results returned by the {{XRCPUDepthInformation/getDepthInMeters(column, row)}} method.
 
-Note: Applications SHOULD NOT attempt to change the contents of {{XRDepthInformation/data}} array as this can lead to incorrect results returned by the {{XRDepthInformation/getDepth(column, row)}} method.
+The {{XRCPUDepthInformation/getDepthInMeters(column, row)}} method can be used to [=obtain depth at coordinates=].  When invoked, the algorithm to <a lt="attempting to access members">access a member</a> MUST be run.
 
-The {{XRDepthInformation/width}} attribute contains width of the {{XRDepthInformation/data}} buffer (i.e. number of columns).
+<div class="algorithm" data-algorithm="obtain-depth-at-coordinates">
 
-The {{XRDepthInformation/height}} attribute contains height of the {{XRDepthInformation/data}} buffer (i.e. number of rows).
+When {{XRCPUDepthInformation/getDepthInMeters(column, row)}} method is invoked on an {{XRCPUDepthInformation}} |depthInformation| with |column|, |row|, the user agent MUST <dfn>obtain depth at coordinates</dfn> by running the following steps:
+    1. Let |view| be the |depthInformation|'s [=XRDepthInformation/view=], |frame| be |view|'s [=XRView/frame=], and |session| be the |frame|'s {{XRFrame/session}}.
+    1. If |column| is greater than or equal to |depthInformation|'s {{XRDepthInformation/width}}, [=exception/throw=] {{NotAllowedError}} and abort these steps.
```

Comment:
Would [`RangeError`](https://heycam.github.io/webidl/#exceptiondef-rangeerror) be more appropriate for these width/height validations than `NotAllowedError`?

Reply:
Yes, thanks!